test/component/Tooltip/Tooltip.sync.spec.tsx

```diff
@@ -0,0 +1,382 @@
+//   name: 'Sankey',
+//   Wrapper: ({ children }) => (
+//     <Sankey width={400} height={400} margin={{ top: 20, right: 20, bottom: 20, left: 20 }} data={SankeyData}>
+//       {children}
+//     </Sankey>
+//   ),
+//   mouseHoverSelector: sankeyNodeChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(35px, 114.89236115144739px);',
+// };
+
+// const SunburstChartTestCase: TooltipVisibilityTestCase = {
+//   name: 'SunburstChart',
+//   Wrapper: ({ children }) => (
+//     <SunburstChart width={400} height={400} data={exampleSunburstData}>
+//       {children}
+//     </SunburstChart>
+//   ),
+//   mouseHoverSelector: sunburstChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(285px, 210px);',
+// };
+
+// const TreemapTestCase: TooltipVisibilityTestCase = {
+//   name: 'Treemap',
+//   Wrapper: ({ children }) => (
+//     <Treemap
+//       width={400}
+//       height={400}
+//       data={exampleTreemapData}
+//       isAnimationActive={false}
+//       nameKey="name"
+//       dataKey="value"
+//     >
+//       {children}
+//     </Treemap>
+//   ),
+//   mouseHoverSelector: treemapNodeChartMouseHoverTooltipSelector,
+//   expectedTransform: 'transform: translate(94.5px, 58.5px);',
+// };
+
+const testCases: ReadonlyArray<TooltipVisibilityTestCase> = [
+  AreaChartTestCase,
+  BarChartTestCase,
+  LineChartHorizontalTestCase,
+  LineChartVerticalTestCase,
+  ComposedChartWithAreaTestCase,
+  ComposedChartWithBarTestCase,
+  ComposedChartWithLineTestCase,
+  // FunnelChartTestCase,
+  // PieChartTestCase,
+  RadarChartTestCase,
+  RadialBarChartTestCase,
+  // ScatterChartTestCase,
+];
+
+// Tooltip sync does not work for PieChart, ScatterChart, FunnelChart, SunburstChart, SankeyChart, Treemap
+describe('Tooltip synchronization', () => {
+  describe.each(testCases)('as a child of $name', ({ name, Wrapper, mouseHoverSelector }) => {
+    test(`${name} shows tooltip when synchronized with ${name}`, () => {
+      const { container, debug } = render(
+        <>
+          <Wrapper syncId="tooltipSync">
+            <Tooltip />
+          </Wrapper>
+          <Wrapper syncId="tooltipSync">
+            <Tooltip />
+          </Wrapper>
+        </>,
+      );
+      const wrappers = container.querySelectorAll('.recharts-wrapper');
+      const wrapperTwo = wrappers[1];
+
+      // target the first chart to show the tooltip
+      showTooltip(container, mouseHoverSelector, debug);
+
+      // target the second chart to see if it has the synchonized tooltip showing
+      const tooltip = getTooltip(wrapperTwo);
+      expect(tooltip).toBeVisible();
+
+      const tooltipContentName = wrapperTwo.querySelector('.recharts-tooltip-item-name');
+      const tooltipContentValue = wrapperTwo.querySelector('.recharts-tooltip-item-value');
+      expect(tooltipContentName).not.toBeNull();
+      expect(tooltipContentValue).not.toBeNull();
+      expect(tooltipContentName).toBeInTheDocument();
```

Comment:
Can we also `expect`:

- the tooltip content (name + value)
- same for the other tooltip? both of them should display, right? and each with different content depending on their respective `dataKey`s

Reply:
Sure. And probably different content given different data, but the same indices